kornia/augmentation/_2d/intensity/salt_pepper_noise.py

```diff
@@ -0,0 +1,133 @@
+
+    def __init__(
+        self,
+        amount: Optional[Union[float, Tuple[float, float]]] = (0.01, 0.06),
+        salt_vs_pepper: Optional[Union[float, Tuple[float, float]]] = (0.4, 0.6),
+        p: float = 0.5,
+        same_on_batch: bool = False,
+        keepdim: bool = False,
+    ) -> None:
+        super().__init__(p=p, same_on_batch=same_on_batch, p_batch=1.0, keepdim=keepdim)
+
+        # Validation and initialization of amount and salt_vs_pepper parameters.
+        if isinstance(salt_vs_pepper, (tuple, float)):
+            if isinstance(salt_vs_pepper, float):
+                salt_vs_pepper = (salt_vs_pepper, salt_vs_pepper)
+            elif len(salt_vs_pepper) == 1:
+                salt_vs_pepper = (salt_vs_pepper[0], salt_vs_pepper[0])
+            elif len(salt_vs_pepper) > 2 or len(salt_vs_pepper) <= 0:
+                raise ValueError(
+                    "The length of salt_vs_pepper must be greater than 0 \
+                        and less than or equal to 2, and it should be a tuple."
+                )
+        else:
+            raise ValueError("salt_vs_pepper must be a tuple or a float")
+        KORNIA_CHECK(
+            all(0 <= el <= 1 for el in salt_vs_pepper),
+            "Salt_vs_pepper values must be between 0 and 1. \
+                        Recommended value 0.5.",
+        )
+
+        if isinstance(amount, (tuple, float)):
+            if isinstance(amount, float):
+                amount = (amount, amount)
+            elif len(amount) == 1:
+                amount = (amount[0], amount[0])
+            elif len(amount) > 2 or len(amount) <= 0:
+                raise ValueError(
+                    "The length of amount must be greater than 0 \
+                        and less than or equal to 2, and it should be a tuple."
+                )
+        else:
+            raise ValueError("amount must be a tuple or a float")
+        KORNIA_CHECK(
+            all(0 <= el <= 1 for el in amount),
+            "amount of noise values must be between 0 and 1. \
+                        Recommended values less than 0.2.",
+        )
+
+        # Generator of random parameters and masks.
+        self._param_generator = SaltAndPepperGenerator(amount, salt_vs_pepper)
+
+    def apply_transform(
+        self,
+        input: Tensor,
+        params: Dict[str, Tensor],
+        flags: Dict[str, Any],
+        transform: Optional[Tensor] = None,
+    ) -> Tensor:
+        r"""Apply random Salt and Pepper noise transformation to the input image."""
+        KORNIA_CHECK(len(input.shape) in (3, 4), "Wrong input dimension.")
+        if len(input.shape) == 3:
+            input = input[None, :, :, :]
+        KORNIA_CHECK(input.shape[1] in {3, 1}, "Number of color channels should be 1 or 3.")
+
+        # Create noisy image cloning input.
+        noisy_image = input.clone()
+
+        # Apply add and multiply mask.
+        noisy_image += params["mask_salt"]
+        noisy_image *= params["mask_pepper"]
```

Comment:
Not sure if it is still a thing. The in-place operations had some issues whilst backpropagation previously.
```suggestion
        noisy_image = noisy_image + params["mask_salt"]
        noisy_image = noisy_image * params["mask_pepper"]
```

Reply:
Is it necessary to clone the image or not? I will replace the in-place operations to avoid problems.